builders/integrations/bridges/wormhole.md

```diff
@@ -0,0 +1,422 @@
+| [Polygon Mumbai](https://faucet.polygon.technology/){target=_blank} | 0x0CBE91CF822c73C2315FB05100C2F714765d5c20 |         5         |
+|    [Avalanche Fuji](https://faucet.avax.network/){target=_blank}    | 0x7bbcE28e64B3F8b84d876Ab298393c38ad7aac4C |         6         |
+|   [Fantom Testnet](https://faucet.fantom.network/){target=_blank}   | 0x1BB3B4119b7BA9dfad76B0545fb3F531383c3bB7 |         10        |
+|         [Goreli](https://goerlifaucet.com/){target=_blank}          | 0x706abc4E45D419950511e474C7B9Ed348A4a716c |         2         |
+| [Moonbase Alpha](https://docs.moonbeam.network/builders/get-started/networks/moonbase/#moonbase-alpha-faucet){target=_blank} | 0xa5B7D85a8f27dd7907dc8FdC21FA5657D5E2F901 | 16 |
+
+1. Ensure that the contract chosen is SimpleGeneralMessage
+2. Open up the deploy menu with the arrow button
+3. Input the relevant *chainID* in the **_CHAINID** input
+4. Input the relevant core bridge address in the **WORMHOLE_CORE_BRIDGE_ADDRESS** input
+5. Press the **transact** button to start a deployment transaction
+6. Press the **Confirm** button in MetaMask to deploy
+
+Once the contract has been deployed on Moonbase Alpha make sure to copy down its address and repeat the process with one of any of the other [EVM TestNets](https://layerzero.gitbook.io/docs/technical-reference/testnet/testnet-addresses){target=_blank} that are connected to Wormhole so that you can send a message across chains. Remember that you will have to change your network in MetaMask to deploy to the right network. 
+
+### Whitelisting Moonbase Alpha’s Connected Contract {:whitelisting-moonbase-alpha-connected-contract}
+
+At this point, you should have the same smart contracts deployed twice. One on Moonbase Alpha, and another on another EVM chain.  
+
+Wormhole recommends to include a whitelisting system in their connected contracts, which you will have to use in *SimpleGeneralMessage* before attempting to send a cross-chain message.
+
+To add a whitelisted contract, you must invoke the *addTrustedAddress* function, which requires a *bytes32* formatted address and a domain ID. You can find the domain ID in the [table above](#deploying-the-wormhole-contract-with-remix-on-moonbase-alpha) and on [Wormhole’s documentation](https://book.wormhole.com/reference/contracts.html#testnet){target=_blank}.
+
+```javascript
+function addTrustedAddress(bytes32 sender, uint16 _chainId) external {
+    myTrustedContracts[sender][_chainId] = true;
+}
+```
+
+Note that the *sender* parameter is a _bytes32_ instead of an _address_. Wormhole’s VAAs provide emitter (origin) addresses in the form of bytes32, so they are stored and checked as _bytes32_. To convert an _address_ type to _bytes32_, you will need to pad an additional 24 zeros. This is because an _address_ value is 20 bytes, less than the 32 for _bytes32_. Every byte has 2 hexadecimal characters, so:
```

Comment:
```suggestion
Note that the *sender* parameter is a _bytes32_ instead of an _address_. Wormhole’s VAAs provide emitter (origin) addresses in the form of _bytes32_, so they are stored and checked as _bytes32_. To convert an _address_ type to _bytes32_, you will need to pad an additional 24 zeros. This is because an _address_ value is 20 bytes, less than the 32 for _bytes32_. Every byte has 2 hexadecimal characters, so:
```